### ISSUE TITLE
Compilation of `trash` feature fails on darwin

### ISSUE THREAD

**Issue by Schweber** [Canop/broot#887]
The [compilation](https://logs.ofborg.org/?key=nixos/nixpkgs.317823&attempt_id=7f9fa173-57a6-4e4b-9c6a-bc8825b16a61) of broot on x86_64_darwin fails with
```
error[E0433]: failed to resolve: could not find `os_limited` in `trash`
   --> src/app/panel_state.rs:162:34
    |
162 |                 let res = trash::os_limited::list()
    |                                  ^^^^^^^^^^ could not find `os_limited` in `trash`
    |
note: found an item that was configured out
   --> /private/tmp/nix-build-broot-1.39.0.drv-0/broot-1.39.0-vendor.tar.gz/trash/src/lib.rs:350:9
    |
350 | pub mod os_limited {
    |         ^^^^^^^^^^
error[E0433]: failed to resolve: could not find `os_limited` in `trash`
   --> src/app/panel_state.rs:164:32
    |
164 |                         trash::os_limited::purge_all(items)
    |                                ^^^^^^^^^^ could not find `os_limited` in `trash`
    |
note: found an item that was configured out
```
In the [readme](https://github.com/Byron/trash-rs) of trash it says:

> The library supports Windows, **macOS**, and all FreeDesktop Trash compliant environments (including GNOME, KDE, XFCE, and more).

Is there something that can be done about it? I would like to enable `trash` as the default build Feature in [nixpkgs](https://github.com/NixOS/nixpkgs/pull/317823).

**Comment by Canop** [comment]
This README is misleading IMO: all the interesting features like listing the content of the trash or purging it are implemented only for Linux and Windows. See https://docs.rs/trash/latest/trash/os_limited/index.html

If somebody finds another crate, or implements the missing features, we could make them available in broot but I can't really work on Mac specific things as I don't own a Mac.

**Comment by Canop** [comment]
Reopening, because adding trash features on Mac is a valid issue, even if I can't work alone on it

**Comment by FrancescElies** [comment]
Related https://github.com/Byron/trash-rs/issues/8.

Long story short there seems to be some API limitations on macos that makes implementing trash operations on mac rather difficult.

**Comment by cmontemuino** [comment]
Is it possible to compile without the trash feature?

**Comment by Canop** [comment]
Yes. If you have downloaded the broot repository, you can exectute `cargo install --locked --path .`

**Comment by Canop** [comment]
The trash cargo feature was removed in broot 1.50: trash support is now compiled according to the platform, and broot builds on macOS (the official Mac binary is built natively). Note that the trash listing screen (:open_trash) isn't available on macOS, as the trash crate doesn't support listing there; sending files to the trash with :trash works. Closing.
